### PR TITLE
fix(amplify-provider-awscloudformation): add slow down on index check

### PR DESCRIPTION
**Pull request by SwaySway**
#### Description of changes

cfn is async to ddb gsi creation slowing the api call can wait for gsi operation to start before
checking a table with potentially wrong info

#### Checklist

- [x] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-provider-awscloudformation/src/iterative-deployment/deployment-manager.ts [hunk 1]
```diff
@@ -239,13 +239,17 @@ export class DeploymentManager {
   };
 
   private waitForIndices = async (stackParams: DeploymentMachineOp) => {
-    if (stackParams.tableNames.length) console.log('\nWaiting for DynamoDB table indices to be ready');
+    if (stackParams.tableNames.length) {
+      console.log('\nWaiting for DynamoDB table indices to be ready');
+      // cfn is async to ddb gsi creation the api can return true before the gsi creation starts
+      await new Promise(res => setTimeout(res, 2000));
```
Comment (ammarkarachi) [inline packages/amplify-provider-awscloudformation/src/iterative-deployment/deployment-manager.ts:245]: Is this timeout good for all cases?

Comment (cjihrig) [inline packages/amplify-provider-awscloudformation/src/iterative-deployment/deployment-manager.ts:245]: This reduces the chance of encountering the race condition, but doesn't remove it, right?

**Review by ammarkarachi — APPROVED** [review]

File: packages/amplify-provider-awscloudformation/src/iterative-deployment/deployment-manager.ts (hunk 1 above)
Comment (SwaySway) [reply packages/amplify-provider-awscloudformation/src/iterative-deployment/deployment-manager.ts:245]: Overall it's good to have a wait time when tableNames are passed in. When tables are deleted it's not included in the list as it doesn't need a wait time.

Comment (SwaySway) [reply packages/amplify-provider-awscloudformation/src/iterative-deployment/deployment-manager.ts:245]: Correct, though afaik there is no method to check that the request was made before checking table status. From testing I can say this does happen in very specific scenarios. So far I was only able to repro this by doing a multiple table GSI update where one of the tables had a GSI removed prior and on the next deployment a GSI was added back with same name.

**Pull request merged**